src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
> If the two keyboards has been configured libinput-side to be on different seats, yes.

Again the same problem. I see this is not fixable easily. I don't really get why the protocol supports different keymaps per window, but not between different keyboards (on the same seat).

> My take on this would be

Seems reasonable, lets start with that.